libyul/AsmParser.cpp

```diff
@@ -138,28 +138,48 @@ Statement Parser::parseStatement()
 		return ExpressionStatement{locationOf(expr), expr};
 	}
 	case Token::Comma:
+	case Token::AssemblyAssign:
 	{
-		// if a comma follows, a multiple assignment is assumed
+		auto const& getIdentifier = [this](ElementaryOperation const& _elementary)
```

Comment:
I think the local function here is still not necessary. What about the following?
```
	{
		std::vector<Identifier> variableNames;

		while (true)
		{
			if (elementary.type() != typeid(Identifier))
				fatalParserError(
					std::string("Variable name must precede \"") +
					currentToken() == Token::Comma ? "," : ":=" +
					"\"" +
					(currentToken() == Token::Comma ? " in multiple assignment." : " in assignment.")
				);

            Identifier&& identifier = boost::get<Identifier>(std::move(elementary));

			if (m_dialect->builtin(identifier.name))
				fatalParserError("Cannot assign to builtin function \"" + identifier.name.str() + "\".");

            variableNames.emplaceBack(std::move(identifier));

            if (currentToken() != Token::Comma)
                break;

            expectToken(Token::Comma);

            elementary = parseElementaryOperation();
		};

		expectToken(Token::AssemblyAssign);

        Assignment assignment = createWithLocation<Assignment>(variableNames.front.location);
        assignment.variableNames = std::move(variableNames);
		assignment.value.reset(new Expression(parseExpression()));
		assignment.location.end = locationOf(*assignment.value).end;
		return Statement{std::move(assignment)};
	}
```